controllers/usersignup/usersignup_controller.go

```diff
@@ -357,13 +357,18 @@ func (r *Reconciler) checkIfMurAlreadyExists(
 
 		if shouldManageSpace(userSignup) {
 			space, created, err := r.ensureSpace(ctx, userSignup, mur, spaceTier)
+			if err == nil {
+				err = r.updateStatusHomeSpace(ctx, userSignup, space.Name)
+				if err != nil {
+					return true, err
+				}
+			}
 			// if there was an error or the space was created then return to complete the reconcile, another reconcile will occur when space is created since this controller watches spaces
 			if err != nil {
 				return true, r.wrapErrorWithStatusUpdate(ctx, userSignup, r.setStatusFailedToCreateSpace, err, "error creating Space")
 			} else if created {
 				return true, nil
 			}
```

Comment:
while this is fine logical-wise, it's pretty hard to read. could you reorganize it to something like this:
```suggestion
			if err != nil {
				return true, r.wrapErrorWithStatusUpdate(ctx, userSignup, r.setStatusFailedToCreateSpace, err, "error creating Space")
			}
			if err := r.updateStatusHomeSpace(ctx, userSignup, space.Name); err != nil {
				return true, err
			}
			// if the space was just created then return to complete the reconcile, another reconcile will occur when space is created since this controller watches spaces
			if created {
				return true, nil
			}
```